src/registrar/services/dns_host_service.py

```diff
@@ -39,21 +43,40 @@ def _find_nameservers_by_zone_id(self, items, x_zone_id):
     def dns_setup(self, domain_name):
         account_name = make_dns_account_name(domain_name)
```

Comment:
I would refactor this, break this up into its parts. It's doing too much. That would also help with the error logging across the domino of exceptions possible. 

Return None vs raise.

Reply:
We've definitely been struggling with how to keep the logic together but not make it do too much. I can see splitting `dns_setup` into `account_setup` and `zone_setup`. I'm not sure how to break each of those down further while maintaining the conditional logic. Can you be more specific about how you'd break it up?

Reply:
This has been a drawn out PR and we keep adding tasks, so I'm going to put dns_setup refactor in[ this ticket](https://github.com/cisagov/manage.get.gov/issues/4304) where we will be moving this functionality to an analyst-triggered action.